fission/src/mirabuf/MirabufInstance.ts

```diff
@@ -178,61 +173,51 @@ class MirabufInstance {
-                        mesh.mesh.indices
-                    ) {
-                        const appearanceOverride = body.appearanceOverride
-                        const material: THREE.Material = WIREFRAME
-                            ? new THREE.MeshStandardMaterial({ wireframe: true, color: 0x000000 })
-                            : appearanceOverride && this._materials.has(appearanceOverride)
-                              ? this._materials.get(appearanceOverride)!
-                              : fillerMaterials[nextFillerMaterial++ % fillerMaterials.length]
-
-                        let materialBodyMap = batchMap.get(material)
-                        if (!materialBodyMap) {
-                            materialBodyMap = new Map<string, [mirabuf.IBody, Array<mirabuf.IPartInstance>]>()
-                            batchMap.set(material, materialBodyMap)
-                        }
-
-                        const partBodyGuid = this.GetPartBodyGuid(definition, body)
-                        let bodyInstances = materialBodyMap.get(partBodyGuid)
-                        if (!bodyInstances) {
-                            bodyInstances = [body, new Array<mirabuf.IPartInstance>()]
-                            materialBodyMap.set(partBodyGuid, bodyInstances)
-                        }
-                        bodyInstances[1].push(instance)
-
-                        if (countMap.has(material)) {
-                            const count = countMap.get(material)!
-                            count.maxInstances += 1
-                            count.maxVertices += mesh.mesh.verts.length / 3
-                            count.maxIndices += mesh.mesh.indices.length
-                        } else {
-                            const count: BatchCounts = {
-                                maxInstances: 1,
-                                maxVertices: mesh.mesh.verts.length / 3,
-                                maxIndices: mesh.mesh.indices.length,
-                            }
-                            countMap.set(material, count)
-                        }
-                    }
+        Object.values(instances).forEach(instance => {
+            const definition = assembly.data!.parts!.partDefinitions![instance.partDefinitionReference!]
+            const bodies = definition?.bodies ?? []
+            bodies.forEach(body => {
+                const mesh = body?.triangleMesh?.mesh
+                if (!mesh?.verts || !mesh.normals || !mesh.uv || !mesh.indices) return
+
+                const appearanceOverride = body.appearanceOverride
+                const material: THREE.Material = WIREFRAME
+                    ? new THREE.MeshStandardMaterial({ wireframe: true, color: 0x000000 })
+                    : appearanceOverride && this._materials.has(appearanceOverride)
+                      ? this._materials.get(appearanceOverride)!
+                      : fillerMaterials[nextFillerMaterial++ % fillerMaterials.length]
```

Comment:
Same as previous, although this one was already here and not part of this particular PR. Not sure if this is just a skill issue but I'm finding this much harder to read compared to having a `if` statement do the assignment.

Reply:
It's not a skill issue, it's just a preference difference i think.